syntax="proto3";

package proto_tc;

import "map_renderer.proto";
import "transport_router.proto";

message Stop {
    uint32 id = 1;
    string name = 2;
    double lat = 3;
    double lng = 4;
}

message Bus {
    string name = 1;
    repeated uint32 route = 2;
    uint32 unique_size = 3;
    int64 length_ = 4;
    double geo_length = 5;
    bool is_roundtrip = 6;
    uint32 last_stop = 7;
}

message Distance {
    uint32 stop_x = 1;
    uint32 stop_y = 2;
    int64 distance = 3; 
}

message TransportCatalogue {
    repeated Bus buses = 1;
    repeated Stop stops = 2;
    repeated Distance distances = 3;
}

message GENERAL_DATA {
    proto_tc.TransportCatalogue tc = 1;
    proto_map_render.RenderSettings renderer_settings = 2;
    proto_tr.TransportRouter tr = 3;
}